crates/red_knot_test/src/lib.rs

```diff
@@ -135,7 +137,7 @@ fn run_test(db: &mut db::Db, test: &parser::MarkdownTest) -> Result<(), Failures
 
             Some(TestFile {
                 file,
-                backtick_offset: embedded.backtick_offset,
+                code_block_dimensions: Box::new(embedded.code_block_dimensions()),
```

Comment:
Nit: Could we simplify this by collecting the dimensions instead of using a `Box<dyn impl Iterator>`. I suspect that the performance difference is minimal.

Reply:
I wanted to make it lazy to avoid counting line numbers for all snippets even if we never emit a test failure, but that might have been an unnecessary optimization, I agree. Will look into it.